tellapart/aurproxy/register/azurelb.py

```diff
@@ -111,7 +111,7 @@ def add_vm_to_load_balancer(self, lb, vm, backend_pool=None):
     if not vm:
         logger.warn('no vm to register!')
         return False
-    bp = self._find_backend_pool(lb, None)
+    bp = self._find_backend_pool(lb, 'aurora')
```

Comment:
I vote this be parameterized so:
* If anyone besides us is using this library it doesn't break them
* We can adjust backend pooling naming without requiring a release of aurproxy

Reply:
mhmm i guess besides from my inherent laziness here my thoughts on not making this parameterized are

> - If anyone besides us is using this library it doesn't break them

i think this would effectively have no change for anyone using aurproxy with a single backend pool on the application gateway like we were. The `_find_backend_pool` method will return the first backend pool reported by the azure api if you provide a backend pool name that doesn't exist. Right now, everyone is forced into this behavior because the only call to this method passes in the hardcoded `None` value.
https://github.com/amperity/aurproxy/blob/44af0944d1efde94e709f3bdd84011f2c30bb19b/tellapart/aurproxy/register/azurelb.py#L360-L366

if someone is relying on this library and they aren't using `aurora` as the name of their backend pool, then they would see the same behavior where aurproxy doesn't see the `aurora` backend pool so it just defaults to the first one as before.

if they're using multiple backend pools, well aurproxy wouldn't have worked for them before.

> - We can adjust backend pooling naming without requiring a release of aurproxy

we're hoping to get off aurproxy pretty soon (like within the next few months), so needing to use another backend pool name for aurproxy seems very unlikely to me. i guess since in my view this is essentially a dead project, it's a better use of our time to just get this in a state that won't break in the intermediate stage where we're migrating to kubernetes.

what are your thoughts on that, do you think im thinking about this the right way?

Reply:
> return the first backend pool reported by the azure api if you provide a backend pool name that doesn't exist

ok that sounds totally fine to me then.

also a quick look at "who else is using this" is a vanishingly small group anyway.